internal/inotify/inotify.go

```diff
@@ -0,0 +1,354 @@
+					if e.IsDir() {
+						// a directory got created in a directory
+						removeddir := filepath.Join(watchedPath, e.path)
+						if watchedPath == n.rootDir {
+							// a namespace_pod directory got deleted, remove watch for this directory
+							glog.V(3).Infof("A namespace_pod got deleted. namespace_pod: %q", removeddir)
+						} else {
+							// a container directory got deleted, remove watch for this directory
+							glog.V(3).Infof("A container got deleted. container: %q", removeddir)
+						}
+						must(n.RemoveWatchForPath(removeddir))
+					} else {
+						// a file got deleted in a directory
+						logfile := filepath.Join(watchedPath, e.path)
+						glog.V(3).Infof("A log file got deleted %q", logfile)
+						// don't need to remove watch for the file because files are watched using IN_ONESHOT, and watch would have got removed when file was closed for writing.
+						must(n.RemoveWatchForPath(logfile))
+					}
+				} else {
+					// there should'nt be anything here because delete notification should come on parent directory
+					glog.Errorf("unrecognized IN_DELETE event. wd: %x, watchedPath: %q, path: %q, Mask: %x", e.Wd, watchedPath, e.path, e.Mask)
+				}
+			case e.IsModify():
+				if e.path != "" {
+					// this should not occur as no IN_MODIFY watch is placed for any directory
+					glog.Errorf("unrecognized IN_MODIFY event. wd: %x, watchedPath: %q, path: %q, Mask: %x", e.Wd, watchedPath, e.path, e.Mask)
+				} else {
+					// a log file got written
+					if glog.V(9) {
+						glog.Infof("logfile %q got written", watchedPath)
+					}
+					if err := UpdateFileSize(watchedPath); err != nil {
+						glog.Errorf("Error in doing stat for file: %q, err: %v", watchedPath, err)
+					}
+					// add a new watch for the file
+					must(n.WatchLogFile(watchedPath))
+				}
+			case e.IsCloseWrite():
+				if e.path != "" {
+					// this should not occur as no IN_CLOSE_WRITE watch is placed for any directory
+					glog.Errorf("unrecognized IN_CLOSE_WRITE event. wd: %x, watchedPath: %q, path: %q, Mask: %x", e.Wd, watchedPath, e.path, e.Mask)
+				} else {
+					// a log file opened for writing got closed
+					glog.V(3).Infof("logfile %q got closed for writing", watchedPath)
+					// update file size last time
+					if err := UpdateFileSize(watchedPath); err != nil {
+						glog.Errorf("Error in doing stat for file: %q, err: %v", watchedPath, err)
+					}
+					// add a new watch for the file because we want an event when it is written to again.
+					// If we avoid adding a watch here, we would need to add a IN_OPEN for its parent directory which leads to a large number of unwanted events
+					must(n.WatchLogFile(watchedPath))
+				}
+			default:
+				glog.Errorf("unhandled event. wd: %x, watchedPath: %q, path: %q, Mask: %x", e.Wd, watchedPath, e.path, e.Mask)
+			}
+
+		case <-tickerCh:
+			/**/
+			sizemtx.Lock()
+			glog.V(0).Infof("sizes(%d): %s\nEventsSent: %d, EventsHandled: %d\n", len(Sizes), func(m map[string]int64) string {
```

Comment:
This looks like we should bump the verbosity since Marshalling is likely an expensive operation.  Should the marshal actually have a if block around the actual work so it only happens when v>X

Reply:
This is for debugging purpose only, will remove this once metrics are added.